src/Hint/NewType.hs

```diff
@@ -138,3 +150,41 @@ dropConsBang x = x
 isUnboxedTuple :: HsType GhcPs -> Bool
 isUnboxedTuple (HsTupleTy _ HsUnboxedTuple _) = True
 isUnboxedTuple _ = False
+
+data WarnNewTypeUn = WarnNewTypeUn
+    { unNewDecl      :: LHsDecl GhcPs
+    , unNewFieldName :: String
+    }
+    | NoWarn
+
+newtypeUn :: LHsDecl GhcPs -> [Idea]
+newtypeUn old
+  | WarnNewTypeUn{unNewDecl, unNewFieldName} <- newTypeWithoutUn old
+  = let
+      note fieldName = [Note $ "Rename field to " <> unNewFieldName]
```

Comment:
What's the purpose of this note? The node usually refers to how the hint isn't quite correct, not just a description of what the hint did. My guess is this hint doesn't need a name.